frame/system/src/offchain.rs

```diff
@@ -17,351 +17,486 @@
+			let call = f(account);
+			self.submit_signed_transaction(account, call)
+		})
+	}
 }
 
-impl<S, C, E> Default for TransactionSubmitter<S, C, E> {
-	fn default() -> Self {
-		Self {
-			_signer: Default::default(),
-		}
+impl<
+	T: SigningTypes + CreateSignedTransaction<LocalCall>,
+	C: AppCrypto<T::Public, T::Signature>,
+	LocalCall,
+> SendSignedTransaction<T, C, LocalCall> for Signer<T, C, ForAll> {
+	type Result = Vec<(Account<T>, Result<(), ()>)>;
+
+	fn send_signed_transaction(
+		&self,
+		f: impl Fn(&Account<T>) -> LocalCall,
+	) -> Self::Result {
+		self.for_all(|account| {
+			let call = f(account);
+			self.submit_signed_transaction(account, call)
+		})
 	}
 }
 
-/// A blanket implementation to simplify creation of transaction signer & submitter in the runtime.
-impl<T, E, S, C, Call> SignAndSubmitTransaction<T, Call> for TransactionSubmitter<S, C, E> where
-	T: crate::Trait,
-	C: CreateTransaction<T, E>,
-	S: Signer<<C as CreateTransaction<T, E>>::Public, <C as CreateTransaction<T, E>>::Signature>,
-	E: ExtrinsicT<Call=Call> + Encode,
-{
-	type Extrinsic = E;
-	type CreateTransaction = C;
-	type Signer = S;
+impl<
+	T: SigningTypes + SendTransactionTypes<LocalCall>,
+	C: AppCrypto<T::Public, T::Signature>,
+	LocalCall,
+> SendUnsignedTransaction<T, LocalCall> for Signer<T, C, ForAny> {
+	type Result = Option<(Account<T>, Result<(), ()>)>;
+
+	fn send_unsigned_transaction<TPayload, F>(
+		&self,
+		f: F,
+		f2: impl Fn(TPayload, T::Signature) -> LocalCall,
+	) -> Self::Result
+	where
+		F: Fn(&Account<T>) -> TPayload,
+		TPayload: SignedPayload<T>, 
+	{
+		self.for_any(|account| {
+			let payload = f(account);
+			let signature= payload.sign::<C>()?;
+			let call = f2(payload, signature);
+			self.submit_unsigned_transaction(call)
+		})
+	}
 }
 
-/// A blanket implementation to use the same submitter for unsigned transactions as well.
-impl<T, E, S, C, Call> SubmitUnsignedTransaction<T, Call> for TransactionSubmitter<S, C, E> where
-	T: crate::Trait,
-	E: ExtrinsicT<Call=Call> + Encode,
-{
-	type Extrinsic = E;
+impl<
+	T: SigningTypes + SendTransactionTypes<LocalCall>,
+	C: AppCrypto<T::Public, T::Signature>,
+	LocalCall,
+> SendUnsignedTransaction<T, LocalCall> for Signer<T, C, ForAll> {
+	type Result = Vec<(Account<T>, Result<(), ()>)>;
+
+	fn send_unsigned_transaction<TPayload, F>(
+		&self,
+		f: F,
+		f2: impl Fn(TPayload, T::Signature) -> LocalCall,
+	) -> Self::Result
+	where
+		F: Fn(&Account<T>) -> TPayload,
+		TPayload: SignedPayload<T> {
+		self.for_all(|account| {
+			let payload = f(account);
+			let signature = payload.sign::<C>()?;
+			let call = f2(payload, signature);
+			self.submit_unsigned_transaction(call)
+		})
+	}
+}
+
+/// Account information used for signing payloads
+pub struct Account<T: SigningTypes> {
+	pub index: usize,
+	pub id: T::AccountId,
+	pub public: T::Public,
+}
+
+impl<T: SigningTypes> Account<T> {
+	pub fn new(index: usize, id: T::AccountId, public: T::Public) -> Self {
```

Comment:
Another missing doc. Please check all the functions in that module (or even add `allow(missing_docs)`)